### PR TITLE
TST: clean up skips and xfails

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
sure

File: pandas/tests/frame/methods/test_rank.py [hunk 1]
```diff
@@ -26,8 +28,10 @@ def method(self, request):
         """
         return request.param
 
+    @td.skip_if_no_scipy
     def test_rank(self, float_frame):
-        rankdata = pytest.importorskip("scipy.stats.rankdata")
```
Comment (WillAyd) [inline pandas/tests/frame/methods/test_rank.py]: Were these ever running before? I guess this isn't directly importable

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @jbrockmendel

File: pandas/tests/frame/methods/test_rank.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/frame/methods/test_rank.py]: AFAICT no.  im working on a branch now to reduce how many skips we have so we can keep a closer eye on them

Comment (WillAyd) [reply pandas/tests/frame/methods/test_rank.py]: Great!

**Pull request merged**